### DISCUSSION TITLE
RPC: install from a specific path

### DISCUSSION THREAD

**Discussion by vincenzopalazzo** [coffee-tools/coffee#185] [Ideas]
There is a use case that we are not covering, and it is when we want to run a local repository, and this means that we should
in some way tell coffee to install the repository from the specific path, so the question now is how.

Just use the install command, like `coffee install --override <new root path>`? or override a remote with a local path? `coffee remote <> <local path>`?

or we can use a combination of the two, but this is still not clear to me

**Comment by tareknaser** [comment]
I think it's better to add the path as a remote repository so then the user can have multiple plugins in the same folder and coffee will crawl the folder to add all of them to the list of plugins available.
It could look like:
```rust
coffee remote add <repo_name> <repo_local_path>
``` 
If we are to add this feature, we also have to keep track of which repositories are cloned from git and which are local repositories. This information will be useful when we want to upgrade the repository for example.

**Reply by vincenzopalazzo** [reply]
yeah but this required also the --override to the install, because now we have to plugin with the same name

This is starting quickly to be a mess :) so we should think for a bit on it

**Reply by tareknaser** [reply]
I agree. Let's not rush this one
